pkg/addon/utils.go

```diff
@@ -409,3 +439,21 @@ func generateAnnotation(meta *Meta) map[string]string {
 func isErrorCueRenderPathNotFound(err error, path string) bool {
 	return err.Error() == fmt.Sprintf("var(path=%s) not exist", path)
 }
+
+func checkConflictDefs(ctx context.Context, k8sClient client.Client, defs []*unstructured.Unstructured, app v1beta1.Application) ([]string, error) {
+	var res []string
+	for _, def := range defs {
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(def), def)
+		if err == nil {
+			owner := metav1.GetControllerOf(def)
+			if owner == nil || owner.Kind != v1beta1.ApplicationKind || owner.Name != app.Name {
+				// if addon not belong to an addon or addon name is another one, we should put them in result
```

Comment:
the condition judgement is really clever! But the explanation here is confusing.

```suggestion
			if owner == nil || owner.Kind != v1beta1.ApplicationKind || owner.Name != app.Name {
				// owner is nil means the definition don't belong to any addons, it must conflict with us.
				// owner.Kind != v1beta1.ApplicationKind or name not equal, it also means the owner is not this addon.
```

BTW, it is possible that Definition has more than 1 owners , what if  you're checking the wrong one?

Reply:
Have consider this situation, because addon-app always is the controller owner, which can only be one.

Reply:
I think it's better to check if this application is one of the addon

Reply:
I think if the addon-app is controller owner of this def, that's mean the addon will modify at any time. Is it  fine to update it directly?